File: app/src/main/java/co/covid19/model/Classifier.java
package co.covid19.model;

import android.app.Activity;
import android.graphics.Bitmap;

import co.covid19.model.exceptions.ImageNotConvertedException;

public interface Classifier<T, V> {
    String getFolderName();
    String getModelPath();
    T getModel(Activity a);
    V imageToTensor(Bitmap b) throws ImageNotConvertedException;
    V getOutputTensor(V inputTensor);
    DiagnosisEnum getResult(V outputTensor);
}

File: app/src/main/java/co/covid19/model/Classifier.kt
package co.covid19.model

import android.graphics.Bitmap

interface Classifier {
    val folderName: String
    val modelPath: String
    var model: <A : Any>
    fun imageToTensor(image: Bitmap?): <T : Any>
    fun loadModelFromPath()
    fun getOutputTensor(inputTensor: <T : Any>): <T: Any>
    fun getResult(outputTensor: <T: Any>): DiagnosisEnum
}